project/apps/api-gateway/src/user/user.controller.ts

```diff
@@ -0,0 +1,132 @@
+  
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  HttpStatus,
+  Param,
+  Patch,
+  Put,
+  Req,
+  Res,
+  Inject,
+} from '@nestjs/common';
+import { ClientProxy } from '@nestjs/microservices';
+import {
+  UpdateUserDto,
+  updateUserSchema
+} from '@repo/dtos/user';
+import { ZodValidationPipe } from '@repo/pipes/zod-validation-pipe.pipe';
+import { Request, Response } from 'express';
+import { firstValueFrom } from 'rxjs';
+@Controller('user')
+export class UserController {
+  constructor(
+    @Inject('USER_SERVICE')
+    private readonly authServiceClient: ClientProxy,
+    private readonly userServiceClient: ClientProxy,
+  ) {}
+  
+  @Get('all-users')
+  async getAllUsers(@Req() request: Request, @Res() res: Response) {
+    const token = request.cookies['access_token'];
+    if (!token) {
+      return res.status(HttpStatus.UNAUTHORIZED).json({ user: null });
+    }
+    
+    // Check if user is admin
+    const { userData } = await firstValueFrom(
+      this.authServiceClient.send({ cmd: 'me' }, token)
+    );
+    if (userData.role !== 'admin') {
+      return res.status(HttpStatus.FORBIDDEN).json({ message: 'Access denied' });
+    }
```

Comment:
FYI ill be abstracting these out to a RoleGuard in a subsequent [PR](https://github.com/CS3219-AY2425S1/cs3219-ay2425s1-project-g30/issues/31)

Reply:
yep okay i'll leave the code with identifying role there first then